src/renderers/WebGLRenderer.js

```diff
@@ -1055,6 +1055,8 @@ function WebGLRenderer( parameters ) {
 
 		currentRenderState.setupLights( camera );
 
+		let visited = {};
```

Comment:
Please use `var`  and a weak map. Otherwise you introduce a memory leak.

Reply:
No memory leak. The object will be ready for garbage collection at the end of the function call, whether it is declared with a `var` or with a `let`. Note that I did specifically avoid trying to extend the memory of the visits beyond the `compile` call.

Reply:
Oh, of course. You're right! However, you should still use `var`.

Reply:
I can use `var`. It makes no difference, but I guess it aids compatibility with Netscape Navigator. ;-)